src/MessageRouter.py

```diff
@@ -0,0 +1,47 @@
+import time
+import re
+
+
+class MessageRouter:
+    """
+    Responsible for invoking a message command handler if a message contains
+    a command and the rate limit is not exceeded.
+    """
+
+    def __init__(self,
+                 command_handlers={},
+                 default_rate_limit_in_seconds=60,
+                 per_command_rate_limit_in_seconds={}):
+        self._default_rate_limit = default_rate_limit_in_seconds
+        self._command_handlers = command_handlers
+        self._per_command_rate_limit = per_command_rate_limit_in_seconds
+
+        self._time_of_last_command_usage = {}
+
+    def handle_message(self, message):
```

Comment:
Do we need a CLI tool for handling irc messages?

SRP violation is a matter of definition. If you define the responsibility of `BotModeration` (Let's rename it to fafbot already) as "be an IRC bot" it is not unreasonable for it to have a single line of code to dispatch messages onto itself, otherwise this line could be defined in a mixin or such.